.kotlin_alpha/usecases/serverless_rds/src/main/kotlin/com/example/demo/DemoApplication.kt

```diff
@@ -0,0 +1,133 @@
+/*
+   Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+   SPDX-License-Identifier: Apache-2.0
+*/
+
+
+package com.example.demo
+
+import kotlinx.coroutines.runBlocking
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.autoconfigure.SpringBootApplication
+import org.springframework.boot.runApplication
+import org.springframework.stereotype.Controller
+import org.springframework.web.bind.annotation.*
+import javax.servlet.http.HttpServletRequest
+import javax.servlet.http.HttpServletResponse
+
+@SpringBootApplication
+class DemoApplication
+
+fun main(args: Array<String>) {
+    runApplication<DemoApplication>(*args)
+}
+
+@Controller
+class MessageResource {
+
+    @Autowired
+    lateinit var injectItems: InjectWorkService
+
+    @Autowired
+    lateinit var sendMsg: SendMessage
+
+    @Autowired
+    lateinit var ri: RetrieveItems
+
+    @GetMapping("/")
+    fun root(): String? {
+        return "index"
+    }
+
+    @GetMapping("/add")
+    fun designer(): String? {
+        return "add"
+    }
+
+    @GetMapping("/items")
+    fun items(): String? {
+        return "items"
+    }
+
+    // Retrieve all items for a given user.
+    @RequestMapping(value = ["/retrieve"], method = [RequestMethod.POST])
+    @ResponseBody
+    fun retrieveItems(request: HttpServletRequest, response: HttpServletResponse?): String?  = runBlocking {
+
+        val type = request.getParameter("type")
+        val name = "user"
+
+        // Pass back all data from the database.
+        val xml: String?
+
+        return@runBlocking if (type == "active") {
+            xml = ri.getItemsDataSQL(name, 0)
+            xml
+        } else {
+            xml = ri.getItemsDataSQL(name, 1)
+            xml
+        }
+    }
+
+    // Adds a new item to the database.
```

Comment:
Adds or Add? In line 52 you use 'Retrieve' rather than 'Retrieves'.
Throughout, when describing what a piece of code does, you alternate between adding an 's' (which denotes present tense and implies 'This adds') and not adding an 's' (which denotes an instruction). I'd suggest making a choice and sticking with it, for consistency.  Either is valid IMHO. My preference is no 's', but its up to you.

Reply:
fixed